Model/Api/Request.php

```diff
@@ -166,4 +166,17 @@ public function sendV1ApiRequest(
         $response->setData('is_success', $successFullResponse);
         return $response;
     }
+
+    /**
+     * @return DataObject
+     */
+    public function getEmptyResponse()
+    {
+        $responseObject = new \Magento\Framework\DataObject();
+        $responseObject->setData('status', null);
+        $responseObject->setData('reason', null);
+        $responseObject->setData('response', []);
+        $responseObject->setData('is_success', false);
```

Comment:
Hi @manaftryzens 
This 4 fields are a mix of two concerns that are split and duplicated:

1. `status`, `reason`, and `response` which being returned from `yotpoHttpclient`
2. `is_success` that is being added by calling `yotpoResponse->validateResponse`

The same logic applied both on `sendV1ApiRequest` and `send`.

Please move the addition of `is_success` to the `yotpoHttpclient`, and remove the duplication.

After that, I assume that the `getEmptyResponse` will need to move to the `yotpoHttpclient` as well.

You can consider [this commit](https://github.com/YotpoLtd/magento2-module-core/pull/71/commits/60f5f06678d7556001031de70917e4974e6bbeee) as a reference for the change (only implemented on `send`, and not on `sendV1ApiRequest`).

Reply:
Hi Eran,

The above commit link seems empty now. It says "We went looking everywhere, but couldn’t find those commits."

Also please note that we are not calling yotpoHttpclient::send() method here. We are just creating an empty response. 

yotpoHttpclient returns status, reason, and response after sending a real http request to the API

Reply:
Hi @manaftryzens ,

- Please give [this one](https://github.com/YotpoLtd/magento2-module-core/commit/ac3d784b69b0b152ee9f43ebb297883e829b1aab) a look.
- I understand that this method doesn't create any network call. I believe it belongs to the Request. I believe that the commit example will make it clear.

Reply:
@EranAvidor ,

The mentioned changes have been implemented. Please review